libs/clients/smartsolutions/src/lib/smartsolutions.module.ts

```diff
@@ -1,8 +1,44 @@
-import { Module } from '@nestjs/common'
-import { SmartSolutionsApi } from './smartSolutions.api'
+import { DynamicModule, Module } from '@nestjs/common'
+import { SmartSolutionsApi, SmartSolutionsConfig } from './smartSolutions.api'
+import { SMART_SOLUTIONS_API_CONFIG } from './smartSolutions.config'
+import { ConfigType, LazyDuringDevScope } from '@island.is/nest/config'
 
-@Module({
-  providers: [SmartSolutionsApi],
-  exports: [SmartSolutionsApi],
-})
-export class SmartSolutionsApiClientModule {}
+export type SmartSolutionsModuleAsyncOptions = {
+  useFactory: (config: ConfigType<never>) => SmartSolutionsConfig
+  inject: Array<string>
+}
```

Comment:
This is closer in line to the SequelizeModule.forRootAsync types:

```suggestion
export type SmartSolutionsModuleAsyncOptions = {
  useFactory: (...args: any[]) => Promise<SmartSolutionsConfig> | SmartSolutionsConfig;
  inject?: any[];
}
```

Note how inject and the useFactory args are optional and under the consumer's control.